plugins/tasks/tasks-core/jira/src/com/intellij/tasks/jira/JiraRepository.java

```diff
@@ -266,10 +270,16 @@ public String executeMethod(@NotNull HttpMethod method) throws Exception {
     // See https://confluence.atlassian.com/display/ONDEMANDKB/Getting+randomly+logged+out+of+OnDemand for details
     // IDEA-128824, IDEA-128706 Use cookie authentication only for JIRA on-Demand
     // TODO Make JiraVersion more suitable for such checks
-    if (BASIC_AUTH_ONLY || !isInCloud()) {
+    if (BASIC_AUTH_ONLY) {
       // to override persisted settings
       setUseHttpAuthentication(true);
     }
+    else if (!isInCloud()) {
+      if (isUseBearerTokenAuthentication()) {
+        setUseHttpAuthentication(false);
+        method.addRequestHeader(new Header(HttpAuthenticator.WWW_AUTH_RESP, "Bearer " + myPassword));
+      }
```

Comment:
I think we still need to have `setUseHttpAuthentication(true)` in an `else` branch here to retain the old behavior when the checkbox is off.

Reply:
good call, added